src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerMovementsEventService.kt

```diff
@@ -20,25 +24,76 @@ class PrisonerMovementsEventService(private val domainEventEmitter: HmppsDomainE
         prisonId = movementChange.prisonId,
         fromPrisonId = null /* TODO: get previous prisonId from movements API */
       )
+
+      is CourtReturn -> domainEventEmitter.emitPrisonerReceiveEvent(
+        offenderNo = movementChange.offenderNo,
+        reason = movementChange.reason,
+        prisonId = movementChange.prisonId,
+      )
+
+      is TAPReturn -> domainEventEmitter.emitPrisonerReceiveEvent(
+        offenderNo = movementChange.offenderNo,
+        reason = movementChange.reason,
+        prisonId = movementChange.prisonId,
+      )
     }
   }
 
   private fun calculateMovementChange(previousPrisonerSnapshot: Prisoner?, prisoner: Prisoner): PossibleMovementChange {
     return previousPrisonerSnapshot.let {
-      // TODO: when we do a few more of these we can extract a nice state machine pattern
-      if (previousPrisonerSnapshot?.inOutStatus == "TRN" && prisoner.inOutStatus == "IN") {
-        TransferIn(prisoner.prisonerNumber!!, prisoner.prisonId!!)
+      val prisonerNumber = prisoner.prisonerNumber!!
```

Comment:
any suggestions for alternative for `if else` is more than welcome

Reply:
You could use `when` - fewer lines but each harder to read. Not much in it.